Sources/SwiftProtobuf/ProtobufJSONTypes.swift

```diff
@@ -818,12 +818,7 @@ public extension ProtobufJSONMessageBase {
     }
 
     func serializeAnyJSON() throws -> String {
-        var jsonVisitor = ProtobufJSONEncodingVisitor()
-        try jsonVisitor.withAbstractVisitor {(visitor: inout ProtobufVisitor) in
-            try visitor.visitSingularField(fieldType: ProtobufString.self, value: anyTypeURL, protoFieldNumber: 1, protoFieldName: "type_url", jsonFieldName: "@type", swiftFieldName: "typeURL")
-            try traverse(visitor: &visitor)
-        }
-        return jsonVisitor.result
+        return try ProtobufJSONEncodingVisitor(message: self, anyTypeURL: anyTypeURL).result
```

Comment:
There are several different pieces here:
  1. Standard JSON serialization of standard messages.
  2. Standard JSON serialization of WKTs
  3. 'Any JSON' serialization of standard messages  (which adds an `@type` field)
  4. 'Any JSON' serialization of WKTs (which adds an `@type` field and puts the custom JSON version into a `value` field)

I initially tried to build one JSON encoding flow that handled all four of these and it got to be a right mess.  I managed to tame it some by treating the last two as a fundamentally separate encoding/decoding flow (which share a lot of the back-end infrastructure with the first two).  That's how `serializeAnyJson()` came to exist.

Where it gets really confusing is the connection between the second item above (regular JSON encoding of Any fields) and the last two (the encoding of the Any field contents).